### DISCUSSION TITLE
I'd like to use RNW for my next windows application, but it's not clicking for me; need some help getting started

### DISCUSSION THREAD

**Discussion by BenMakesGames** [microsoft/react-native-windows#8427] [General]
I have an Angular background, and love component-based frameworks. I haven't used React before, but it feels pretty natural to me, and I'm excited to use it.

However, I'm unable to figure out how to structure an RNW application with a heavy C# backend (Entity Framework, etc), and I can't find many resources online (as to be expected, since RNW is so new).

My intuition is to structure a Windows application as follows:
1. An entry point - static void Main - which loads configuration, registers DI/IoC, Entity Framework DB context, logging, etc, using `Microsoft.Extensions.*` libraries
2. As its final step, this entry point passes off to the front-end: React Native
3. C# services, db context, etc, are available to the React front-end via DI

I know that the above won't work as written: React is the starting point, already has DI, etc. Knowing that, I assume I need to register my C# services with React somehow. I've tried exporting service classes from RNW native modules using `[ReactModule]` and friends, but this makes the compiler unhappy: apparently native module classes need parameterless constructors. In retrospect, this makes sense: React JS and native C# services aren't going to be able to seamlessly depend on one another... but I need DI on the C# side, for the services defined there... and so I find myself stuck in a loop/trap.

I feel as though I'm missing some bigger picture which would make everything fall into place, I'm just not sure where/how to find it. Any help would be greatly appreciated!

**Comment by chrisglein** [comment]
You'll have to manually surface your C# logic to your React Native JS through native modules, using '[ReactModule]` as you've raised (documentation [here](https://microsoft.github.io/react-native-windows/docs/0.64/native-modules), more samples [here](https://github.com/microsoft/react-native-windows-samples)). You said that made the compiler unhappy due to constructor parameters. Was that the only compiler issue? Is that a requirement for your types? Have you had luck replicating any of the examples in the docs/samples?

**Reply by BenMakesGames** [reply]
constructor parameters aren't a requirement... just changes how I architect the solution. I haven't used RNW before, or even RN, so I think I just have to start over, in some senses, when thinking about how things are organized/architected.

I'm currently focusing my investigations on Blazor Desktop, but if that doesn't pan out, I'll be back for RNW :P

thanks for getting back!

**Comment by asklar** [comment]
Communication between apps and modules can happen via the ReactPropertyBag object, which is an instance wide property bag.
